src/hotspot/share/gc/shenandoah/shenandoahUnload.cpp

```diff
@@ -77,8 +78,7 @@ class ShenandoahIsUnloadingOopClosure : public OopClosure {
 
 class ShenandoahIsUnloadingBehaviour : public IsUnloadingBehaviour {
 public:
-  virtual bool has_dead_oop(CompiledMethod* method) const {
-    nmethod* const nm = method->as_nmethod();
+  virtual bool has_dead_oop(nmethod* const nm) const {
```

Comment:
Is there a reason why this was changed to `nmethod* const nm` instead of `nmethod* nm`? IsUnloadingBehviour::has_dead_oop uses `nmethod* nm`. This question applies to the other changes in this file as well.